lib/src/Uncertainty/Algorithm/MetaModel/FunctionalChaos/openturns/FunctionalChaosValidation.hxx

```diff
@@ -0,0 +1,97 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief The validation of a chaos expansion
+ *
+ *  Copyright 2005-2023 Airbus-EDF-IMACS-ONERA-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#ifndef OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+#define OPENTURNS_FUNCTIONALCHAOSVALIDATION_HXX
+
+#include "openturns/Sample.hxx"
+#include "openturns/FunctionalChaosResult.hxx"
+#include "openturns/MetaModelValidation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+/**
+ * @class FunctionalChaosValidation
+ *
+ * The validation of a chaos expansion
+ */
+
+class OT_API FunctionalChaosValidation
+  : public MetaModelValidation
+{
+  CLASSNAME
+
+public:
+  typedef Collection<Function>           FunctionCollection;
+  enum CrossValidationMethod {LEAVEONEOUT = 0, KFOLD = 1};
+
+  /** Default constructor */
+  FunctionalChaosValidation();
+
+  /** Parameter constructor */
+  FunctionalChaosValidation(const FunctionalChaosResult & chaosResult,
+                            const CrossValidationMethod method, 
+                            const UnsignedInteger & kParameter = ResourceMap::GetAsUnsignedInteger("FunctionalChaosValidation-DefaultKFoldParameter"));
+
+  /** Virtual constructor */
+  FunctionalChaosValidation * clone() const override;
+
+  /** String converter */
+  String __repr__() const override;
+
+  /** Result accessor */
+  FunctionalChaosResult getFunctionalChaosResult() const;
+  
+  /** Get the K-parameter */
+  UnsignedInteger getKParameter() const;
+
+  /** Get the CV Method */
+  UnsignedInteger getMethod() const;
+
+  /** Method save() stores the object through the StorageManager */
+  void save(Advocate & adv) const override;
+
+  /** Method load() reloads the object from the StorageManager */
+  void load(Advocate & adv) override;
+
+protected:
+
+private:
+
+  /** The functional chaos result */
+  FunctionalChaosResult functionalChaosResult_;
+
+  /** K-parameter */
+  UnsignedInteger kParameter_;
```

Comment:
lets initialize it here

Reply:
What value would you want to set here? `ResourceMap::GetAsUnsignedInteger("FunctionalChaosValidation-DefaultKFoldParameter")`?